### DISCUSSION TITLE
Voice and video calls - which server is being used for that? Are they encrypted?

### DISCUSSION THREAD

**Discussion by M-Stenzel** [processone/ejabberd#4055] [Questions]
According to

[https://www.process-one.net/blog/how-to-set-up-ejabberd-video-voice-calling/]

I setup calls in ejabberd. Now I wonder, which server is being used for this functionality?
Is it the very same server that ejabberd runs on? I only found about STUN/TURN but this only relates to
establishing a connection between parties.

I was not able to find information regarding encryption of voice and video calls. Are they encrypted by default?
I do not see any options in the ejabberd.yml file.

Thank you very much for clarification,

Martin.

**Comment by licaon-kter** [comment]
You setup ejabberd for calls... so the server that **you** setup will be used, be it the same ejabberd instance or an external one (ejabberd or eturnal or coturn or...)

STUN is about finding out users IPs. for the connection.

TURN is about routing that connection through **your** server that **you** setup, when the two clients can't directly connect.

**Comment by weiss** [comment]
> Are they encrypted by default?

Audio/video calls are typically end-to-end encrypted these days, WebRTC [mandates DTLS-SRTP][1], for example.  However, this is a client-side functionality.

[1]: https://webrtc-security.github.io/
